inst/stan/functions/spatial_deviation_noise_matrix_rng.stan

```diff
@@ -0,0 +1,31 @@
+/**
+  * Generates a matrix where columns are random vectors for spatial process.
+  * Currently this is for an exponential correaltion function.
+  * @param dist_matrix Distance matrix for correlation constr.
+  * @param sigma_eps Parameter for construction of covariance matrix
+  * of spatial epsilon.
+  * @param phi Phi parameter for exponential decay corr. func.
+  * @param l l paraamter for exponential decay corr. func.
+  * @param n_time timepoints for number of random vectors of spatial process desired.
+  * @return A matrix for spatial deviation noise in use of spatial Rt process.
+  */
+matrix spatial_deviation_noise_matrix_rng(matrix dist_matrix,
```

Comment:
If these function is going to be specifically for exponential decay, should indicate that in the function name. Alternatively (and I think this is my preference), you could rewrite this as a more generic function that takes in an abitrary`sigma_matrix`. One could then compute the `sigma_matrix` first and then pass it to this function.

If you wanted an exponential decay rng, you could then construct it by calling this more generic function with a `sigma_matrix` computed using exponential decay.